### ISSUE TITLE
Use of nested route groups with 'prefix' and 'as' parameters results in duplicates in route name on Route::resource only

### ISSUE THREAD

**Issue by SirLamer** [laravel/framework#14547]
When you nest Route groups using the properties 'prefix' and 'as', both the 'prefix' AND 'as' properties are use to build the route name only on Route::resource and not Route::get and others. This results in inconsistent patterning when building out complex routes.

Example is below, tested in Laravel v 5.2.41. Explicit route model binding is used in this example but not shown here.

```php
    Route::group(['namespace' => 'Admin', 'middleware' => ['role:admin']], function()
    {
        Route::group(['prefix' => 'admin', 'as' => 'admin.'], function() {
            Route::get('', 'AdminController@index')->name('index');
            Route::group(['prefix' => 'projects', 'as' => 'projects.'], function() {
                // Task Type (part of Admin UI)
                Route::group(['prefix' => 'tasks', 'as' => 'tasks.'], function () {
                    Route::get('defaults', 'TaskTypeController@defaults')->name('defaults');
                    Route::post('defaults', 'TaskTypeController@defaultsUpdate')->name('defaults');
                });
                Route::resource('tasks', 'TaskTypeController', [
                    'parameters' => ['tasks' => 'task_type'],
                ]);
                Route::get('test', function() {
                    return 'lol';
                })->name('test');
            });
        });
    });
```

```
|        | GET|HEAD  | admin                                                             | admin.index                                 | App\Http\Controllers\Admin\AdminController@index               | web,auth,role:admin          |
|        | POST      | admin/projects/tasks                                              | admin.projects.admin.projects.tasks.store   | App\Http\Controllers\Admin\TaskTypeController@store            | web,auth,role:admin          |
|        | GET|HEAD  | admin/projects/tasks                                              | admin.projects.admin.projects.tasks.index   | App\Http\Controllers\Admin\TaskTypeController@index            | web,auth,role:admin          |
|        | GET|HEAD  | admin/projects/tasks/create                                       | admin.projects.admin.projects.tasks.create  | App\Http\Controllers\Admin\TaskTypeController@create           | web,auth,role:admin          |
|        | GET|HEAD  | admin/projects/tasks/defaults                                     | admin.projects.tasks.defaults               | App\Http\Controllers\Admin\TaskTypeController@defaults         | web,auth,role:admin          |
|        | POST      | admin/projects/tasks/defaults                                     | admin.projects.tasks.defaults               | App\Http\Controllers\Admin\TaskTypeController@defaultsUpdate   | web,auth,role:admin          |
|        | DELETE    | admin/projects/tasks/{task_type}                                  | admin.projects.admin.projects.tasks.destroy | App\Http\Controllers\Admin\TaskTypeController@destroy          | web,auth,role:admin          |
|        | GET|HEAD  | admin/projects/tasks/{task_type}                                  | admin.projects.admin.projects.tasks.show    | App\Http\Controllers\Admin\TaskTypeController@show             | web,auth,role:admin          |
|        | PUT|PATCH | admin/projects/tasks/{task_type}                                  | admin.projects.admin.projects.tasks.update  | App\Http\Controllers\Admin\TaskTypeController@update           | web,auth,role:admin          |
|        | GET|HEAD  | admin/projects/tasks/{task_type}/edit                             | admin.projects.admin.projects.tasks.edit    | App\Http\Controllers\Admin\TaskTypeController@edit             | web,auth,role:admin          |
|        | GET|HEAD  | admin/projects/test                                               | admin.projects.test                         | Closure                                                        | web,auth,role:admin          |

```

**Comment by SirLamer** [comment]
I guess I should say... I THINK this is an issue? I just can't see why you would want this behavior.

It also happens while not nesting, however in that case you can get around it a bit more easily. When you do have nested grouping then it's a lot more harmful to clarity in order to fix it.

**Comment by fernandobandeira** [comment]
This should be fixed in 5.3 since prefix isn't appending to the route name anymore.

**Comment by SirLamer** [comment]
Thanks, I don't have a working copy of 5.3 available to check but it sounds like it's fixed.
